superset/dashboards/dao.py

```diff
@@ -238,6 +238,8 @@ def set_dash_metadata(  # pylint: disable=too-many-locals
                 if int(key) in slice_ids
             }
             md["default_filters"] = json.dumps(applicable_filters)
+        # positions have its own column, no need to store it in metadata
+        md.pop("positions", None)
```

Comment:
Maybe we should do a migartion instead?

Reply:
`positions` still need to be passed to the function as certain logics are happening here that use it. What it does not need is to be stored as its own object, reason why it's now being popped out. 

In general, this function has some pretty mixed logic and it would require a bigger refactor. Things like the title, the css, and other data that is unrelated to the dashboard metadata are being used here when they shouldn't. Let's plan for a full refactor later.

Reply:
Thanks for clarification!